hpcgap/src/gvars.c

```diff
@@ -146,11 +160,16 @@ void UnlockGVars() {
 **  will return the value of <gvar>  after evaluating <gvar>-s expression, or
 **  0 if <gvar> was not an automatic variable.
 **
-**  'VAL_GVAR' is defined in the declaration part of this package as follows
-**
-#define VAL_GVAR(gvar)          (PtrGVars[GVAR_BUCKET(gvar)] \
-                                [GVAR_INDEX(gvar)-1])
 */
+#undef VAL_GVAR
+#define VAL_GVAR(gvar)          (PtrGVars[GVAR_BUCKET(gvar)] \
```

Comment:
`VAL_GVAR` and `ValGVar` are not the same in HPC-GAP -- the latter contains a `MEMBAR_READ()`.

So perhaps what you suggest is the right thing to do -- but I did not feel competent to quickly judge this, and hence decided to defer this until a latter stage; I'd rather not put a change into a refactoring commit which might have major consequences.

If it is the #undef/redefine dance you dislike (I actually do, too): I could rename the "second" `VAL_GVAR` defined here to, say `VAL_GVAR_RAW` or `VAL_GVAR_INTERN`, and perhaps add a FIXME explaining that somebody should look into whether we need it.... Or perhaps you have another suggestion?